File: src/test/resources/lib/oidc-test.js
const test = require('/lib/xp/testing');
const testUtils = require('/testUtils');

let capturedRequest = null;

testUtils.mockAndGetUpdaterFunc('/lib/http-client', {
    request: (request) => {
        capturedRequest = request;
        return { status: 500 };
    },
});

const oidc = require('/lib/oidc');

function tokenParams(extra) {
    const params = {
        issuer: 'https://provider.example.com',
        tokenUrl: 'https://provider.example.com/token',
        clientId: 'my-client',
        clientSecret: 's3cr3t',
        redirectUri: 'https://app.example.com/callback',
        nonce: 'nonce-value',
        code: 'auth-code',
        idProviderName: 'oidc',
    };
    Object.keys(extra || {}).forEach(key => { params[key] = extra[key]; });
    return params;
}

exports.testBasicAuthEncodesCredentials = () => {
    capturedRequest = null;
    const result = oidc.requestIDToken(tokenParams({ method: 'basic' }));

    test.assertEquals(500, result.status);
    test.assertEquals('Basic bXktY2xpZW50OnMzY3IzdA==', capturedRequest.headers.Authorization);
};

exports.testBasicAuthEncodesCredentialsAsUtf8 = () => {
    capturedRequest = null;
    const result = oidc.requestIDToken(tokenParams({ method: 'basic', clientId: 'user', clientSecret: 'pÿ€' }));

    test.assertEquals(500, result.status);
    test.assertEquals('Basic dXNlcjpww7/igqw=', capturedRequest.headers.Authorization);
};

File: src/test/java/com/enonic/app/oidcidprovider/lib/OidcTest.java
package com.enonic.app.oidcidprovider.lib;

import org.junit.jupiter.api.Test;

import com.enonic.xp.testing.ScriptTestSupport;

public class OidcTest
    extends ScriptTestSupport
{
    private void run( final String function )
    {
        runFunction( "/lib/oidc-test.js", function );
    }

    @Test
    public void testBasicAuthEncodesCredentials()
    {
        run( "testBasicAuthEncodesCredentials" );
    }

    @Test
    public void testBasicAuthEncodesCredentialsAsUtf8()
    {
        run( "testBasicAuthEncodesCredentialsAsUtf8" );
    }
}
